heat/core/tests/test_statistics.py

```diff
@@ -176,6 +178,64 @@ def test_argmin(self):
         with self.assertRaises(ValueError):
             ht.argmin(data, axis=-4)
 
+    def test_cov(self):
+        x = ht.array([[0, 2], [1, 1], [2, 0]], dtype=ht.float, split=1).T
+        if x.comm.size < 3:
+            cov = ht.cov(x)
+            actual = ht.array([[1, -1], [-1, 1]], split=0)
+            self.assertTrue(ht.equal(cov, actual))
+
+        f = h5py.File('heat/datasets/data/iris.h5', 'r')
+        data = f['data']
+        np_cov = np.cov(data[:, 0], data[:, 1:3], rowvar=False)
+
+        htdata = ht.load('heat/datasets/data/iris.h5', 'data', split=0)
+        ht_cov = ht.cov(htdata[:, 0], htdata[:, 1:3], rowvar=False)
+        self.assertTrue(ht.allclose(ht.array(np_cov, dtype=ht.float) - ht_cov, 0, atol=1e-4))
+
+        np_cov = np.cov(data, rowvar=False)
+        ht_cov = ht.cov(htdata, rowvar=False)
+        self.assertTrue(ht.allclose(ht.array(np_cov, dtype=ht.float) - ht_cov, 0, atol=1e-4))
+
+        np_cov = np.cov(data, rowvar=False, ddof=1)
+        ht_cov = ht.cov(htdata, rowvar=False, ddof=1)
+        self.assertTrue(ht.allclose(ht.array(np_cov, dtype=ht.float) - ht_cov, 0, atol=1e-4))
+
+        np_cov = np.cov(data, rowvar=False, bias=True)
+        ht_cov = ht.cov(htdata, rowvar=False, bias=True)
+        self.assertTrue(ht.allclose(ht.array(np_cov, dtype=ht.float) - ht_cov, 0, atol=1e-4))
+
+        if 1 < x.comm.size < 5:
```

Comment:
why is this condition here? the functions should work in principle for all node combinations

Reply:
its because with the split=1 condition there is an empty tensor on the nodes > 4. this throws an error from the binary_op
```
Traceback (most recent call last):
[4]   File "testing.py", line 46, in <module>
[4]     ht_cov = ht.cov(htdata, rowvar=False)
[4]   File "/home/d.coquelin/.git/heat/heat/core/statistics.py", line 419, in cov
[4]     x -= avg[:, None]
[4]   File "/home/d.coquelin/.git/heat/heat/core/dndarray.py", line 2643, in __sub__
[4]     return arithmetics.sub(self, other)
[4]   File "/home/d.coquelin/.git/heat/heat/core/arithmetics.py", line 374, in sub
[4]     return operations.__binary_op(torch.sub, t1, t2)
[4]   File "/home/d.coquelin/.git/heat/heat/core/operations.py", line 85, in __binary_op
[4]     output_shape = stride_tricks.broadcast_shape(t1.shape, t2.shape)
[4]   File "/home/d.coquelin/.git/heat/heat/core/stride_tricks.py", line 48, in broadcast_shape
[4]     raise ValueError('operands could not be broadcast, input shapes {} {}'.format(shape_a, shape_b))
[4] ValueError: operands could not be broadcast, input shapes (4, 150) (4,)
```

Reply:
interesting, is this documented as an issue?